driver-core/src/test/java/com/datastax/driver/core/CCMTestsSupport.java

```diff
@@ -0,0 +1,687 @@
+        if (testMode == PER_METHOD)
+            initTestContext(testInstance, testMethod);
+    }
+
+    @AfterMethod(groups = {"isolated", "short", "long"}, alwaysRun = true)
+    public void tearDown() {
+        if (testMode == PER_METHOD)
+            closeTestContext();
+    }
+
+    @AfterClass(groups = {"isolated", "short", "long"}, alwaysRun = true)
+    public void cleanUp() {
+        if (testMode == PER_CLASS)
+            closeTestContext();
+    }
+
+
+    /**
+     * Returns the CCMBridge builder to use for this test.
+     * <p/>
+     * The default implementation returns a standard CCM cluster configuration;
+     * user-defined functions are enabled for protocol versions >= 4.
+     *
+     * @return The CCM config to use for the tests.
+     */
+    public CCMBridge.Builder createCCMBridgeBuilder() {
+        CCMBridge.Builder builder = CCMBridge.builder()
+                .withoutNodes()
+                .notStarted();
+        ProtocolVersion protocolVersion = TestUtils.getDesiredProtocolVersion();
+        if (protocolVersion.toInt() >= 4) {
+            builder = builder.withCassandraConfiguration("enable_user_defined_functions", "true");
+        }
+        return builder;
+    }
+
+    /**
+     * Returns the cluster builder to use for this test.
+     * <p/>
+     * The default implementation returns a vanilla builder.
+     *
+     * It's not required to call {@link com.datastax.driver.core.Cluster.Builder#addContactPointsWithPorts},
+     * it will be done automatically.
+     *
+     * @return The cluster builder to use for the tests.
+     */
+    public Cluster.Builder createClusterBuilder() {
+        return Cluster.builder();
+    }
+
+    /**
+     * Returns an alternate cluster builder to use for this test,
+     * with no event debouncing.
+     * <p/>
+     * It's not required to call {@link com.datastax.driver.core.Cluster.Builder#addContactPointsWithPorts},
+     * it will be done automatically.
+     *
+     * @return The cluster builder to use for the tests.
+     */
+    @SuppressWarnings("unused")
+    public Cluster.Builder createClusterBuilderNoDebouncing() {
+        return Cluster.builder()
+                .withQueryOptions(new QueryOptions()
+                        .setRefreshNodeIntervalMillis(0)
+                        .setRefreshNodeListIntervalMillis(0)
+                        .setRefreshSchemaIntervalMillis(0));
+    }
+
+
+    /**
+     * The CQL statements to execute before the tests in this class.
+     * Useful to create tables or other objects inside the test keyspace,
+     * or to insert test data.
+     * <p/>
+     * Statements do not need to be qualified with keyspace name.
+     * <p/>
+     * The default implementation returns an empty list (no DDL statements required).
+     *
+     * @return The DDL statements to execute before the tests.
+     */
+    public Collection<String> createTestFixtures() {
```

Comment:
The previous version of this method (getTableDefinitions) was abstract.  I think I like defaulting this but wanted to point out the difference.